src/fft/domain.rs

```diff
@@ -14,35 +14,71 @@
 
 use super::Evaluations;
 use crate::error::Error;
-use core::fmt;
 use dusk_bls12_381::{BlsScalar, GENERATOR, ROOT_OF_UNITY, TWO_ADACITY};
+use dusk_bytes::{DeserializableSlice, Serializable};
 use rayon::iter::{IndexedParallelIterator, IntoParallelRefMutIterator, ParallelIterator};
 use std::ops::MulAssign;
 
 /// Defines a domain over which finite field (I)FFTs can be performed. Works
 /// only for fields that have a large multiplicative subgroup of size that is
 /// a power-of-2.
-#[derive(Copy, Clone, Eq, PartialEq)]
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
 pub struct EvaluationDomain {
     /// The size of the domain.
-    pub size: u64,
+    pub(crate) size: u64,
     /// `log_2(self.size)`.
-    pub log_size_of_group: u32,
+    pub(crate) log_size_of_group: u32,
     /// Size of the domain as a field element.
-    pub size_as_field_element: BlsScalar,
+    pub(crate) size_as_field_element: BlsScalar,
     /// Inverse of the size in the field.
-    pub size_inv: BlsScalar,
+    pub(crate) size_inv: BlsScalar,
     /// A generator of the subgroup.
-    pub group_gen: BlsScalar,
+    pub(crate) group_gen: BlsScalar,
     /// Inverse of the generator of the subgroup.
-    pub group_gen_inv: BlsScalar,
+    pub(crate) group_gen_inv: BlsScalar,
     /// Multiplicative generator of the finite field.
-    pub generator_inv: BlsScalar,
+    pub(crate) generator_inv: BlsScalar,
 }
 
-impl fmt::Debug for EvaluationDomain {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "Multiplicative subgroup of size {}", self.size)
+impl Serializable<{ u64::SIZE + u32::SIZE + 5 * BlsScalar::SIZE }> for EvaluationDomain {
+    type Error = dusk_bytes::Error;
+
+    #[allow(unused_must_use)]
```

Comment:
What is being unused here?

Reply:
The results from `writer.write()`.

Reply:
Maybe we should put a comment there explaining the situation?